src/c++/perf_analyzer/data_loader.cc

```diff
@@ -329,23 +331,27 @@ DataLoader::GetInputData(
         *batch1_size = string_data->size();
       }
       *data_ptr = (const uint8_t*)&((it->second)[0]);
-    } else {
-      return cb::Error(
-          "unable to find data for input '" + input.name_ +
-          "' in provided data.");
+      data_found = true;
     }
-  } else if (
-      (input.datatype_.compare("BYTES") != 0) && (input_buf_.size() != 0)) {
-    int64_t byte_size = ByteSize(input.shape_, input.datatype_);
-    if (byte_size < 0) {
-      return cb::Error(
-          "failed to get correct byte size for '" + input.name_ + "'.");
+  }
+
+  if (!data_found) {
+    if ((input.datatype_.compare("BYTES") != 0) && (input_buf_.size() != 0)) {
+      int64_t byte_size = ByteSize(input.shape_, input.datatype_);
+      if (byte_size < 0) {
+        return cb::Error(
+            "failed to get correct byte size for '" + input.name_ + "'.");
+      }
+      *batch1_size = (size_t)byte_size;
```

Comment:
Just curiosity, do we ever encounter issues casting `int64_t` -> `size_t` ? Do we support 32-bit systems or other scenarios where `size_t` might be smaller than an `int64_t`?

Reply:
Good observation. I don't think we support any 32-bit machines as of yet. The `size_t`  and `int64_t` is used interchangeably all over the places.